nmaipy/parcels.py

```diff
@@ -429,163 +437,135 @@ def feature_attributes(
+                for key, val in primary_attributes.items():
+                    parcel[f"primary_{name}_" + str(key)] = val
+
+            if class_id == ROOF_ID:
+                if not calc_buffers:
+                    continue
+
+                assert parcel_geom is not None, "Parcel geometry must be provided for buffer calculations"
+
+                # Convert everything to area-based CRS upfront
+                area_crs = AREA_CRS[country]
+                parcel_geom_area = gpd.GeoSeries([parcel_geom], crs=LAT_LONG_CRS).to_crs(area_crs)[0]
+                primary_roof_geom_area = gpd.GeoSeries([primary_feature.geometry_feature], crs=LAT_LONG_CRS).to_crs(area_crs)[0]
+
+                # Get geodataframe of only the relevant features for buffering, and convert to area projection
+                buffer_features_gdf = gpd.GeoDataFrame(
+                    features_gdf[features_gdf.class_id.isin(BUFFER_CLASSES.values())],
+                    geometry="geometry_feature",
+                    crs=LAT_LONG_CRS
+                    )
+                buffer_features_gdf = buffer_features_gdf.to_crs(area_crs)
+
+                # Calculate buffers for each distance
+                for buffer_name, buffer_dist in BUFFER_ZONES_M.items():
+                    # Create column names
+                    parcel[f"primary_roof_{buffer_name}_zone_sqm"] = None
+                    parcel[f"primary_roof_{buffer_name}_all_buffer_classes_sqm"] = None
+                    for bc_name in BUFFER_CLASSES.keys():
+                        parcel[f"primary_roof_{buffer_name}_{bc_name}_sqm"] = None
+
+                for buffer_name, buffer_dist in BUFFER_ZONES_M.items():
+                    # Create buffered regions around roofs
+                    buffered_primary_roof_geom_area = primary_roof_geom_area.buffer(buffer_dist)
+
+                    if not buffered_primary_roof_geom_area.within(parcel_geom_area):
+                        # Skip if the buffer protrudes outside the parcel
+                        continue
+
+                    # Proceed calculating intersections etc. with the buffered primary roof
+                    parcel[f"primary_roof_{buffer_name}_zone_sqm"] = buffered_primary_roof_geom_area.area
+
+                    # Trim buffer features to only the buffer zone, and drop any that are empty
+                    bf_trimmed_gdf = buffer_features_gdf.copy()
+                    bf_trimmed_gdf["geometry_feature"] = bf_trimmed_gdf.intersection(buffered_primary_roof_geom_area)
+                    bf_trimmed_gdf = bf_trimmed_gdf[~bf_trimmed_gdf.is_empty]
+
+                    # Calculate total area of union of all buffer features
+                    buffer_union = bf_trimmed_gdf.union_all()
+                    parcel[f"primary_roof_{buffer_name}_all_buffer_classes_sqm"] = buffer_union.area
+
+                    # TODO: Figure out what to do about the interior of the primary roof - is it part of the zone?
```

Comment:
@mbewley good question - we need to exclude interior of primary roof - it is not part of the zone.

Reply:
As per these results, there should be a 0ft buffer, which is the same as an "overhang" style class.